src/server/api/controllers/routes.controller.js

```diff
@@ -0,0 +1,58 @@
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+const moment = require("moment-timezone");
+
+const createRouteWithWaypoints = async ({ body }) => {
+  const {
+    port_id,
+    voyage_id,
+    eta,
+    max_wave_height,
+    hfo,
+    lsfo,
+    total_cost,
+    distance_over_ground,
+    distance_through_water,
+    avgspeed,
+    waypoints
+  } = body;
```

Comment:
There is a small thing with naming conventions here. I would like to use camelCase in javascript.

In the database we use snake_case and in paths/urls and in filenames we use hypen-case (or kebab-case as some people call it?).